src/coreclr/jit/importer.cpp

```diff
@@ -8231,6 +8231,35 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
         }
         break;
 
+        case CORINFO_FIELD_STATIC_DATASEGMENT:
+        {
+#ifdef FEATURE_READYTORUN
+            assert(opts.IsReadyToRun());
+            assert((pFieldInfo->fieldFlags & CORINFO_FLG_FIELD_INITCLASS) == 0);
+            
+            if (pFieldInfo->fieldLookup.accessType == InfoAccessType::IAT_VALUE)
+            {
+                op1 = gtNewIconHandleNode((size_t)pFieldInfo->fieldLookup.addr, GTF_ICON_STATIC_HDL);
+                op1->gtType = TYP_BYREF;
+            }
+            else
+            {
+                assert(pFieldInfo->fieldLookup.accessType == InfoAccessType::IAT_PVALUE);
+                op1 = gtNewIndOfIconHandleNode(TYP_BYREF, (size_t)pFieldInfo->fieldLookup.addr, GTF_ICON_GLOBAL_PTR, false);
```

Comment:
> `GTF_ICON_GLOBAL_PTR`, `isInvariant: false`

What is the scenario where the address of the static can change?

Note that some code in the Jit depends on the addresses of statics being invariant. It will not matter for this change now, I believe, but I have changes in progress that will make this dependency more visible.

(To be clear: mutable addresses can absolutely be made to work, I just need to understand the scenarios I will need to add as tests to account for them)

Reply:
gtNewIndOfIconHandleNode asserts if GTF_ICON_GLOBAL_PTR is used with isInvariant = true. This was just me hammering on it until it worked. Maybe we need a new `GTF_` value for this? I'm not particularly experienced in the RyuJIT code base; I know just enough to CTRL-C/CTRL-V from various places.

Reply:
Yes, a new handle kind would be the best solution (`GTF_STATIC_ADDR_PTR` say...). Using `GTF_ICON_CONST_PTR` will work too but is less desirable. When adding a new handle kind, please also add printing for it to `cTreeFlags` and `gtDispConst`.

<details>
<summary>Why a new handle kind?</summary>

A "raw" `GTF_ICON_CONST_PTR` will work as an invariant indirection, but a dedicated handle kind will allow VN to treat specially (and more precisely):
```cs
// We will have staticPtr = IND(STATIC_ADDR_PTR) and give "staticPtr" a "VNF_PtrToStatic" VN.
ref var staticPtr = ref staticField;

staticPtr = { ... }; // The compiler will be able to see that this store is just "staticField = { ... }"
```

It will be very similar to `GTF_ICON_STATIC_BOX_PTR`.
</details>